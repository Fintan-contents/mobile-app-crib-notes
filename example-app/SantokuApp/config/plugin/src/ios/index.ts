export * from './withIosRemoveCFBundleUrlTypes';
export * from './withIosEnabledATS';
export * from './withIosDisabledATS';
export * from './withIosSetCredentials';
export * from './withIosAddPersonalAccountConfig';
export * from './withIosOverrideStoryboard';
