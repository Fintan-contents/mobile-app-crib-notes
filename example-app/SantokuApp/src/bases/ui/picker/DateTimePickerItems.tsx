import RNDateTimePicker, {AndroidNativeProps, IOSNativeProps} from '@react-native-community/datetimepicker';
import React from 'react';

export type DateTimePickerItemsIOSProps = IOSNativeProps;
export type DateTimePickerItemsAndroidProps = AndroidNativeProps;
export type DateTimePickerItemsProps = DateTimePickerItemsIOSProps | DateTimePickerItemsAndroidProps;

export const DateTimePickerItems = (props: DateTimePickerItemsProps) => {
  return <RNDateTimePicker {...props} />;
};
