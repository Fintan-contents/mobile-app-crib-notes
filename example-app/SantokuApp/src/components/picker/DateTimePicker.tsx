import {log} from 'framework/logging';
import React from 'react';
import {Platform, TextInputProps as RNETextInputProps, ViewProps} from 'react-native';

import {DateTimePickerItemsProps} from './DateTimePickerItems';
import {DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdropProps} from './PickerBackdrop';
import {PickerContainerProps} from './PickerContainer';

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;

export type DateTimePickerProps = {
  /**
   * 選択されたアイテムのKey
   */
  selectedValue?: Date;
  /**
   * 選択されたアイテムが存在しない場合のデフォルト値
   */
  defaultValue?: Date;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (date?: Date) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   */
  onDismiss?: (date?: Date) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDelete?: (date?: Date) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onCancel?: (date?: Date) => void;
  /**
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDone?: (date?: Date) => void;
  /**
   * 選択した日時のフォーマッタ
   */
  formatText?: (date?: Date) => string | undefined;
  /**
   * 選択可能な最大日時
   * @see https://github.com/react-native-datetimepicker/datetimepicker#maximumdate-optional
   */
  maximumDate?: Date;
  /**
   * 選択可能な最小日時
   * @see https://github.com/react-native-datetimepicker/datetimepicker#minimumdate-optional
   */
  minimumDate?: Date;
  /**
   * 時刻の表示を24時間表記にするかどうか
   * @see https://github.com/react-native-datetimepicker/datetimepicker#is24hour-optional-windows-and-android-only
   */
  is24Hour?: boolean;
  /**
   * Pickerのモード
   * @see https://github.com/react-native-datetimepicker/datetimepicker#mode-optional
   */
  mode?: DateTimePickerItemsProps['mode'];
  /**
   * Pickerの表示方法
   * このコンポーネントでは、spinnerとinlineのみ使用できます。
   * @see https://github.com/react-native-datetimepicker/datetimepicker#display-optional
   * @platform ios
   */
  displayIOS?: Exclude<DateTimePickerItemsProps['display'], 'clock' | 'calendar' | 'compact' | 'default'>;
  /**
   * Pickerの表示方法
   * @see https://github.com/react-native-datetimepicker/datetimepicker#display-optional
   * @platform android
   */
  displayAndroid?: Exclude<DateTimePickerItemsProps['display'], 'compact' | 'inline'>;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、{@link DefaultPickerAccessory}がデフォルトで表示されます。
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、{@link SelectPickerItems}がデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   * @see https://github.com/react-native-datetimepicker/datetimepicker#component-props--params-of-the-android-imperative-api
   */
  pickerItemsProps?: Omit<
    DateTimePickerItemsProps,
    'value' | 'onChange' | 'mode' | 'display' | 'maximumDate' | 'minimumDate'
  >;
  /**
   * PickerBackdropのProps
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

export const DateTimePicker = (props: DateTimePickerProps) => {
  React.useEffect(() => {
    log.warn(`DateTimePicker is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
