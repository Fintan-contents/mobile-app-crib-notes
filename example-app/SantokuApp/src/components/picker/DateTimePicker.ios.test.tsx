import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {StyleSheet, TextInputProps, View, ViewProps, ViewStyle} from 'react-native';

import {DateTimePicker} from './DateTimePicker.ios';
import {DateTimePickerItemsIOSProps} from './DateTimePickerItems';
import {PickerBackdropProps} from './PickerBackdrop';
import {PickerContainerProps} from './PickerContainer';

jest.doMock('react-native/Libraries/Utilities/Dimensions', () => ({
  get: jest.fn().mockReturnValue({width: 400, height: 1000}),
  set: jest.fn(),
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}));

describe('DateTimePicker only with required props', () => {
  it('renders successfully if invisible', () => {
    const sut = render(
      <DateTimePicker
        textInputProps={{testID: 'textInput'}}
        pickerBackdropProps={{pressableProps: {testID: 'pickerBackdrop'}}}
        pickerContainerProps={{testID: 'pickerContainer'}}
        pickerAccessoryProps={{containerProps: {testID: 'pickerAccessory'}}}
        pickerItemsContainerProps={{testID: 'pickerItemsContainer'}}
      />,
    );

    expect(sut).toMatchSnapshot('DateTimePicker if invisible.');

    const pickerBackdrop = sut.queryByTestId('pickerBackdrop');
    const pickerContainer = sut.queryByTestId('pickerContainer');
    const pickerAccessory = sut.queryByTestId('pickerAccessory');
    const pickerItemsContainer = sut.queryByTestId('pickerItemsContainer');
    const textInput = sut.queryByTestId('textInput');
    expect(pickerBackdrop).toBeNull();
    expect(pickerContainer).toBeNull();
    expect(pickerAccessory).toBeNull();
    expect(pickerItemsContainer).toBeNull();
    expect(textInput).not.toBeNull();
  });

  it('renders successfully if visible', () => {
    // selectedValueに固定値を入れないと、Snapshot取得時に毎回値が変わってしまうので、OptionalPropsですが指定しています。
    const selectedValue = new Date('2022-05-30T00:00:00.000Z');
    // formatTextを指定しないと、タイムゾーンの表記が環境によって違うので、どの環境でも変わらないように指定しています。
    const formatText = (date?: Date) => (date ? date.toISOString() : '');
    const sut = render(
      <DateTimePicker
        selectedValue={selectedValue}
        textInputProps={{testID: 'textInput'}}
        pickerBackdropProps={{pressableProps: {testID: 'pickerBackdrop'}}}
        pickerContainerProps={{testID: 'pickerContainer'}}
        pickerAccessoryProps={{containerProps: {testID: 'pickerAccessory'}}}
        pickerItemsContainerProps={{testID: 'pickerItemsContainer'}}
        pickerItemsProps={{testID: 'pickerItems'}}
        formatText={formatText}
      />,
    );
    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);

    expect(sut).toMatchSnapshot('DateTimePicker if visible.');

    const pickerBackdrop = sut.queryByTestId('pickerBackdrop');
    const pickerContainer = sut.queryByTestId('pickerContainer');
    const pickerAccessory = sut.queryByTestId('pickerAccessory');
    const pickerItemsContainer = sut.queryByTestId('pickerItemsContainer');
    const textInput = sut.queryByTestId('textInput');
    expect(pickerBackdrop).not.toBeNull();
    expect(pickerContainer).not.toBeNull();
    expect(pickerAccessory).not.toBeNull();
    expect(pickerItemsContainer).not.toBeNull();
    expect(textInput).not.toBeNull();
  });
});

describe('DateTimePicker with default value', () => {
  it('defaultValue should be set at open if selectedValue does not exist,', () => {
    const defaultValue = new Date('2022-05-10T00:00:00.000Z');
    const sut = render(<DateTimePicker defaultValue={defaultValue} pickerItemsProps={{testID: 'pickerItems'}} />);
    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);
    expect(sut.getByTestId('pickerItems').props.date).toBe(defaultValue.getTime());
  });
});

describe('DateTimePicker with all props', () => {
  it('should be applied all properly with default xxx component', () => {
    const onSelectedItemChange = jest.fn();
    const onDismiss = jest.fn();
    const onDelete = jest.fn();
    const onCancel = jest.fn();
    const onDone = jest.fn();
    const formatText = jest.fn();
    const selectedValue = new Date('2022-01-01T00:00:00.000Z');
    const maximumDate = new Date('2022-05-31T00:00:00.000Z');
    const minimumDate = new Date('2017-01-01T00:00:00.000Z');
    const sut = render(
      <DateTimePicker
        selectedValue={selectedValue}
        maximumDate={maximumDate}
        minimumDate={minimumDate}
        defaultValue={maximumDate}
        onSelectedItemChange={onSelectedItemChange}
        onDismiss={onDismiss}
        placeholder="please select..."
        textInputProps={{style: {color: 'red'}, testID: 'textInput'}}
        pickerItemsContainerProps={{pointerEvents: 'none', testID: 'pickerItemsContainer'}}
        pickerItemsProps={{style: {backgroundColor: 'yellow'}, testID: 'pickerItems'}}
        pickerBackdropProps={{
          modalProps: {testID: 'pickerBackdropModal'},
          pressableProps: {testID: 'pickerBackdropPressable', style: {backgroundColor: 'green', borderColor: 'red'}},
        }}
        pickerContainerProps={{style: {backgroundColor: 'yellow', borderColor: 'black'}, testID: 'pickerContainer'}}
        pickerAccessoryProps={{
          containerProps: {style: {backgroundColor: 'blue'}, testID: 'pickerAccessory'},
          deleteLabel: 'delete',
          cancelLabel: 'cancel',
          doneLabel: 'done',
          deleteTouchableContainerProps: {testID: 'deleteLink'},
          cancelTouchableContainerProps: {testID: 'cancelLink'},
          doneTouchableContainerProps: {testID: 'doneLink'},
        }}
        onDelete={onDelete}
        onCancel={onCancel}
        onDone={onDone}
        formatText={formatText}
        mode="date"
        displayIOS="spinner"
      />,
    );

    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);

    expect(sut).toMatchSnapshot('DateTimePicker all properly with default xxx component.');

    // assert pickerContainer
    const pickerContainer = sut.getByTestId('pickerContainer');
    const pickerContainerProps = pickerContainer.props as PickerContainerProps;
    expect(pickerContainerProps.style).toEqual({
      backgroundColor: 'yellow',
      borderColor: 'black',
      transform: [{translateY: 1000}],
    });

    // assert pickerItemsContainer
    const pickerItemsContainer = sut.getByTestId('pickerItemsContainer');
    const pickerItemsContainerProps = pickerItemsContainer.props as ViewProps;
    expect(pickerItemsContainerProps.pointerEvents).toBe('none');

    // assert PickerItems
    const pickerItems = sut.getByTestId('pickerItems');
    const pickerItemsProps = pickerItems.props as DateTimePickerItemsIOSProps;
    expect(pickerItemsProps.date).toBe(selectedValue.getTime());
    expect(pickerItemsProps.maximumDate).toBe(maximumDate.getTime());
    expect(pickerItemsProps.minimumDate).toBe(minimumDate.getTime());
    expect(pickerItemsProps.mode).toBe('date');
    // displayはdisplayIOSというPropでレンダリングされます
    // @ts-ignore
    // @ts-ignore
    expect(pickerItemsProps.displayIOS).toBe('spinner');
    const pickerItemsStyle = pickerItemsProps.style as ViewStyle;
    const flattenPickerItemsStyle = StyleSheet.flatten(pickerItemsStyle);
    expect(flattenPickerItemsStyle.backgroundColor).toBe('yellow');

    // assert textInput
    const textInput = sut.getByTestId('textInput');
    const textInputProps = textInput.props as TextInputProps;
    expect(textInputProps.style).toEqual({color: 'red'});

    // assert pickerBackdrop
    const pickerBackdropPressable = sut.getByTestId('pickerBackdropPressable');
    const pickerBackdropModal = sut.getByTestId('pickerBackdropModal');
    const pickerBackdropProps = pickerBackdropPressable.props as PickerBackdropProps;
    fireEvent(pickerBackdropModal, 'onRequestClose');
    expect(pickerBackdropProps.style).toEqual({
      backgroundColor: 'green',
      borderColor: 'red',
      opacity: 0.4,
      bottom: 0,
      display: 'flex',
      left: 0,
      position: 'absolute',
      right: 0,
      top: 0,
    });
    expect(onDismiss).toHaveBeenCalledTimes(1);

    fireEvent.press(pressableContainer);

    // assert pickerAccessory
    const pickerAccessory = sut.getByTestId('pickerAccessory');
    const pickerAccessoryProps = pickerAccessory.props as ViewProps;
    expect(pickerAccessoryProps.style).toEqual({
      backgroundColor: 'blue',
      flexDirection: 'row',
      justifyContent: 'flex-end',
      paddingHorizontal: 10,
      paddingVertical: 8,
    });
    const deleteLink = sut.getByTestId('deleteLink');
    const deleteText = sut.queryByText('delete');
    expect(deleteText).not.toBeNull();
    fireEvent.press(deleteLink);
    expect(onDelete).toHaveBeenCalledTimes(1);
    fireEvent.press(pressableContainer);
    const cancelLink = sut.getByTestId('cancelLink');
    const cancelText = sut.queryByText('cancel');
    expect(cancelText).not.toBeNull();
    fireEvent.press(cancelLink);
    expect(onCancel).toHaveBeenCalledTimes(1);
    fireEvent.press(pressableContainer);
    const doneLink = sut.getByTestId('doneLink');
    const doneText = sut.queryByText('done');
    expect(doneText).not.toBeNull();
    fireEvent.press(doneLink);
    expect(onDone).toHaveBeenCalledTimes(1);

    // formatText
    expect(formatText).toHaveBeenCalledTimes(1);
  });

  it('should be applied all properly with custom xxx component', () => {
    const CustomPickerAccessory = <View testID="customPickerAccessory" />;
    const CustomPicker = <View testID="customPicker" />;
    const CustomTextInput = <View testID="customTextInput" />;
    const selectedValue = new Date('2022-01-01T00:00:00.000Z');
    const maximumDate = new Date('2022-05-31T00:00:00.000Z');
    const minimumDate = new Date('2017-01-01T00:00:00.000Z');
    const onSelectedItemChange = jest.fn();
    const onDismiss = jest.fn();
    const formatText = jest.fn();
    const sut = render(
      <DateTimePicker
        selectedValue={selectedValue}
        maximumDate={maximumDate}
        minimumDate={minimumDate}
        defaultValue={maximumDate}
        onSelectedItemChange={onSelectedItemChange}
        onDismiss={onDismiss}
        placeholder="please select..."
        textInputProps={{testID: 'defaultTextInput'}}
        pickerItemsContainerProps={{pointerEvents: 'none', testID: 'pickerItemsContainer'}}
        pickerBackdropProps={{
          modalProps: {testID: 'pickerBackdropModal'},
          pressableProps: {testID: 'pickerBackdropPressable', style: {backgroundColor: 'green', borderColor: 'red'}},
        }}
        pickerContainerProps={{style: {backgroundColor: 'yellow', borderColor: 'black'}, testID: 'pickerContainer'}}
        pickerAccessoryProps={{containerProps: {testID: 'defaultPickerAccessory'}}}
        pickerAccessoryComponent={CustomPickerAccessory}
        pickerItemsComponent={CustomPicker}
        textInputComponent={CustomTextInput}
        formatText={formatText}
        mode="date"
        displayIOS="spinner"
      />,
    );
    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);

    expect(sut).toMatchSnapshot('DateTimePicker all properly with custom xxx component.');

    // assert pickerBackdrop
    const pickerBackdropPressable = sut.getByTestId('pickerBackdropPressable');
    const pickerBackdropModal = sut.getByTestId('pickerBackdropModal');
    const pickerBackdropProps = pickerBackdropPressable.props as PickerBackdropProps;
    fireEvent(pickerBackdropModal, 'onRequestClose');
    expect(pickerBackdropProps.style).toEqual({
      backgroundColor: 'green',
      borderColor: 'red',
      opacity: 0.4,
      bottom: 0,
      display: 'flex',
      left: 0,
      position: 'absolute',
      right: 0,
      top: 0,
    });
    expect(onDismiss).toHaveBeenCalledTimes(1);

    fireEvent.press(pressableContainer);

    // assert pickerContainer
    const pickerContainer = sut.getByTestId('pickerContainer');
    const pickerContainerProps = pickerContainer.props as PickerContainerProps;
    expect(pickerContainerProps.style).toEqual({
      backgroundColor: 'yellow',
      borderColor: 'black',
      transform: [{translateY: 1000}],
    });

    // assert pickerAccessory
    const defaultPickerAccessory = sut.queryByTestId('defaultPickerAccessory');
    const customPickerAccessory = sut.queryByTestId('customPickerAccessory');
    expect(defaultPickerAccessory).toBeNull();
    expect(customPickerAccessory).not.toBeNull();

    // assert pickerItemsContainer
    const pickerItemsContainer = sut.getByTestId('pickerItemsContainer');
    const pickerItemsContainerProps = pickerItemsContainer.props as ViewProps;
    expect(pickerItemsContainerProps.pointerEvents).toBe('none');

    // assert textInput
    const defaultTextInput = sut.queryByTestId('defaultTextInput');
    const customTextInput = sut.queryByTestId('customTextInput');
    expect(defaultTextInput).toBeNull();
    expect(customTextInput).not.toBeNull();
  });
});
