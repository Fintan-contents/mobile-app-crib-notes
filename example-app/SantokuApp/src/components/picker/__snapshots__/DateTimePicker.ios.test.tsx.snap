// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker only with required props renders successfully if invisible: DateTimePicker if invisible. 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="pressableContainer"
>
  <View
    pointerEvents="box-only"
  >
    <TextInput
      editable={false}
      testID="textInput"
    />
  </View>
</View>
`;

exports[`DateTimePicker only with required props renders successfully if visible: DateTimePicker if visible. 1`] = `
Array [
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "display": "flex",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
        entering={
          FadeIn {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 500,
            "randomizeDelay": false,
          }
        }
        exiting={
          FadeOut {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 500,
            "randomizeDelay": false,
          }
        }
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.4)",
            "flex": 1,
          }
        }
        testID="pickerBackdrop"
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
      >
        <View
          animatedStyle={
            Object {
              "value": Object {},
            }
          }
          collapsable={false}
          entering={
            SlideInDown {
              "build": [Function],
              "callbackV": [Function],
              "durationV": 500,
              "randomizeDelay": false,
            }
          }
          exiting={
            SlideOutDown {
              "build": [Function],
              "callbackV": [Function],
              "durationV": 500,
              "randomizeDelay": false,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "white",
            }
          }
          testID="pickerContainer"
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "paddingHorizontal": 10,
                "paddingVertical": 8,
              }
            }
            testID="pickerAccessory"
          />
          <View
            testID="pickerItemsContainer"
          >
            <RNDateTimePicker
              date={1653868800000}
              displayIOS="spinner"
              enabled={true}
              mode="date"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 216,
                }
              }
              testID="pickerItems"
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="pressableContainer"
  >
    <View
      pointerEvents="box-only"
    >
      <TextInput
        editable={false}
        testID="textInput"
        value="2022-05-30T00:00:00.000Z"
      />
    </View>
  </View>,
]
`;

exports[`DateTimePicker with all props should be applied all properly with custom xxx component: DateTimePicker all properly with custom xxx component. 1`] = `
Array [
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="modal"
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "display": "flex",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
        entering={
          FadeIn {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 500,
            "randomizeDelay": false,
          }
        }
        exiting={
          FadeOut {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 500,
            "randomizeDelay": false,
          }
        }
        style={
          Object {
            "backgroundColor": "green",
            "borderColor": "red",
            "flex": 1,
          }
        }
        testID="pickerBackdrop"
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
      >
        <View
          animatedStyle={
            Object {
              "value": Object {},
            }
          }
          collapsable={false}
          entering={
            SlideInDown {
              "build": [Function],
              "callbackV": [Function],
              "durationV": 500,
              "randomizeDelay": false,
            }
          }
          exiting={
            SlideOutDown {
              "build": [Function],
              "callbackV": [Function],
              "durationV": 500,
              "randomizeDelay": false,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "yellow",
              "borderColor": "black",
            }
          }
          testID="pickerContainer"
        >
          <View
            testID="customPickerAccessory"
          />
          <View
            pointerEvents="none"
            testID="pickerItemsContainer"
          >
            <View
              testID="customPicker"
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="pressableContainer"
  >
    <View
      pointerEvents="box-only"
    >
      <View
        testID="customTextInput"
      />
    </View>
  </View>,
]
`;

exports[`DateTimePicker with all props should be applied all properly with default xxx component: DateTimePicker all properly with default xxx component. 1`] = `
Array [
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="modal"
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "display": "flex",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
        entering={
          FadeIn {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 500,
            "randomizeDelay": false,
          }
        }
        exiting={
          FadeOut {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 500,
            "randomizeDelay": false,
          }
        }
        style={
          Object {
            "backgroundColor": "green",
            "borderColor": "red",
            "flex": 1,
          }
        }
        testID="pickerBackdrop"
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
      >
        <View
          animatedStyle={
            Object {
              "value": Object {},
            }
          }
          collapsable={false}
          entering={
            SlideInDown {
              "build": [Function],
              "callbackV": [Function],
              "durationV": 500,
              "randomizeDelay": false,
            }
          }
          exiting={
            SlideOutDown {
              "build": [Function],
              "callbackV": [Function],
              "durationV": 500,
              "randomizeDelay": false,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "yellow",
              "borderColor": "black",
            }
          }
          testID="pickerContainer"
        >
          <View
            style={
              Object {
                "backgroundColor": "blue",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "paddingHorizontal": 10,
                "paddingVertical": 8,
              }
            }
            testID="pickerAccessory"
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="deleteLink"
            >
              <Text
                style={
                  Object {
                    "color": "#d9545e",
                    "fontWeight": "bold",
                    "paddingHorizontal": 10,
                  }
                }
              >
                delete
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="cancelLink"
            >
              <Text
                style={
                  Object {
                    "color": "#4577CC",
                    "fontWeight": "bold",
                    "paddingHorizontal": 10,
                  }
                }
              >
                cancel
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="doneLink"
            >
              <Text
                style={
                  Object {
                    "color": "#4577CC",
                    "fontWeight": "bold",
                    "paddingHorizontal": 10,
                  }
                }
              >
                done
              </Text>
            </View>
          </View>
          <View
            pointerEvents="none"
            testID="pickerItemsContainer"
          >
            <RNDateTimePicker
              date={1640995200000}
              displayIOS="spinner"
              enabled={true}
              maximumDate={1653955200000}
              minimumDate={1483228800000}
              mode="date"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  Object {
                    "backgroundColor": "yellow",
                  },
                ]
              }
              testID="pickerItems"
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="pressableContainer"
  >
    <View
      pointerEvents="box-only"
    >
      <TextInput
        editable={false}
        placeholder="please select..."
        style={
          Object {
            "color": "red",
          }
        }
        testID="textInput"
      />
    </View>
  </View>,
]
`;
